community/kernel/src/main/java/org/neo4j/kernel/impl/api/dbms/NonTransactionalDbmsOperations.java

```diff
@@ -32,18 +33,23 @@
 {
 
     private final Procedures procedures;
+    private final KernelTransaction transaction;
 
-    public NonTransactionalDbmsOperations( Procedures procedures )
+    public NonTransactionalDbmsOperations( Procedures procedures, KernelTransaction transaction )
     {
         this.procedures = procedures;
+        this.transaction = transaction;
     }
 
     @Override
     public RawIterator<Object[],ProcedureException> procedureCallDbms( ProcedureSignature.ProcedureName name,
-            Object[] input, AccessMode accessMode ) throws ProcedureException
+            Object[] input ) throws ProcedureException
```

Comment:
This design was explored initially, but the current one was favoured as it is more similar to how read/write/schema-operations work, and exposes the KernelTransaction in fewer places.